src/blog/_posts/2023-07-10-cadence-bad-practices-part-1.md

```diff
@@ -0,0 +1,25 @@
+---
+title: Bad practices and Anti-patterns with Cadence (Part 1)
+
+date: 2023-07-10
+author: Chris Qin
+authorlink: https://www.linkedin.com/in/chrisqin0610/
+---
+
+In the upcoming blog series, we will delve into a discussion about common bad practices and anti-patterns related to Cadence. As diverse teams often encounter distinct business use cases, it becomes imperative to address the most frequently reported issues in Cadence workflows. To provide valuable insights and guidance, the Cadence team has meticulously compiled these common challenges based on customer feedback.
+
+* Reuse same workflow ID for very active/continuous running workflows
+
+At first glance, the practice of maximizing workflow code reusability may seem like a good idea. However, it can cause significant performance problems. This is because Cadence organizes workflows based on their unique IDs, using a process called partitioning. If a workflow receives a large number of updates in a short period or frequently starts new runs using the `continueAsNew` function, all these updates will be directed to the same shard. Unfortunately, the Cadence backend is not equipped to handle this concentrated workload efficiently. As a result, a situation known as a "hot shard" arises, overloading the Cadence backend and worsening the problem.
+
+Solution:
+Well, the best way to avoid this is simply just design your workflow in the way such that each workflow owns a uniformly distributed workflow ID across your Cadence domain. This will make sure that Cadence backend is able to evenly distribute the traffic with proper partition on your workflowIDs.
+
+* Excessive batch jobs or enormous number of timers triggered at the same time
```

Comment:
I've not seen enormous used like this but it's probably OK - needs an in front e.g 'an enormous.....'

Reply:
done